### PR TITLE
Add decorators

### PR DESCRIPTION
**Pull request by IsmaelMastronardi**
## On this pull request : 
- Created a Nameable class with a correct_name method that raises a NotImplementedError.

- Changed Person class to inherit from Nameable and implemented a correct_name method.

- Created a Base Decorator, that inherits from Nameable, assigns a Nameable object to an instance variable, and implements a correct_name method that returns the result from Nameable.

- Created a Capitalize Decorator that capitalizes the output of @nameable.correct_name.

- Created a Trimmer Decorator that returns the output of @nameable.correct_name or trims it to that amount if it is longer than 10 letters.

-To fix the issue with the super raised by the last reviewer, I added a () parenthesis to the super, as the supper is required by lenders, but, as commented by the reviewer, they returned an error as they were written.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
-class Person
+require './nameable'
+
+class Person < Nameable
   attr_reader :id
   attr_accessor :age, :name
 
   def initialize(age, name = 'unknown', parent_permission: true)
+    super
```
Comment (Owusu-Desmond) [inline person.rb]: - [ ] To fix this issue in your project, please remove the `super` keyword from your initialize method. The initialize method calls the parent class's initialize method with the same arguments, but your parent class does not have an initialize method. This caused an error when I tested your project. Simply removing the `super` keyword will resolve this problem.
  
  image

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require './nameable'
+
+class Basedecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super
```
Comment (Owusu-Desmond) [inline base_decorator.rb]: - [ ] To fix this issue in your project, please remove the `super` keyword from your initialize method. The initialize method calls the parent class's initialize method with the same arguments, but your parent class does not have an initialize method. This caused an error when I tested your project. Simply removing the `super` keyword will resolve this problem. :blush: 
    
  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @IsmaelMastronardi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @IsmaelMastronardi , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done fixing the error that was preventing your code from running. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**